packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1221,6 +1230,42 @@ function updateTransition(
   return [start, isPending];
 }
 
+let clientId: number = 0;
+
+function mountOpaqueReference(): string | IdObject {
+  const [id, setId] = mountState(() => {
+    if (getIsHydrating()) {
+      return {
+        $$typeof: REACT_OPAQUE_OBJECT_TYPE,
```

Comment:
Other than its creation, this object is only ever referenced in `ReactDOM`. That's a good thing because on other platforms like the React Native use case, this object will look very different and have a very different signature.

To do that, though, we need to make the creation of this object something platform specific. Let's make a HostConfig method to create this object and only implement it in ReactDOM.

For React Native, we should make that method throw since this feature is not yet supported there. Later we could add some specific implementation.

You probably need to pass `setId` in as an argument to the `HostConfig` method.